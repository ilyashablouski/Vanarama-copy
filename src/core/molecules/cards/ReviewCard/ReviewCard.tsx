import cx from 'classnames';
import React, { FC, memo } from 'react';

import { IBaseProps } from '../../../interfaces/base';
import Rating from '../../../atoms/rating';
import Text from '../../../atoms/text';

interface IReviewCard {
  text: string;
  author: string;
  score: number;
}

export interface IReviewCardProps extends IBaseProps {
  review: IReviewCard;
  optimisedHost?: string;
}

const ReviewCard: FC<IReviewCardProps> = memo(props => {
  const { className, review } = props;
  return (
    <div
      className={cx('card -overflow', {
        [`${className}`]: className,
      })}
      data-testid="card-review"
    >
      <Text
        color="dark"
        size="regular"
        dataTestId="review"
        className="review-text"
      >
        {review.text}
      </Text>
      <div className="review-author" data-testid="author-review">
        {review.author}
      </div>
      <div className="review-decoration">
        <Rating score={review.score} />
      </div>
    </div>
  );
});

export default ReviewCard;
