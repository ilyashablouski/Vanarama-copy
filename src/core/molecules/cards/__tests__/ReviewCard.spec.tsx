import React from 'react';
import { shallow } from 'enzyme';
import ReviewCard from '../ReviewCard/ReviewCard';

describe('<ReviewCard />', () => {
  it('should render correctly', () => {
    // ACT
    const wrapper = shallow(
      <ReviewCard
        review={{
          text:
            'Cillum sit et in nostrud occaecat est labore ea laborum voluptate magna eu aliquip mollit',
          author: 'Review Card',
          score: 4.5,
        }}
      />,
    );

    // ASSERT
    expect(wrapper).toMatchSnapshot();
  });
});
