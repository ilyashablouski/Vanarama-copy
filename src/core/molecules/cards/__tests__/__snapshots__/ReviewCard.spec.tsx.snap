// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCard /> should render correctly 1`] = `
<div>
  <div
    class="card -overflow"
    data-testid="card-review"
  >
    <span
      class="text review-text -regular -dark"
      data-testid="review"
    >
      Cillum sit et in nostrud occaecat est labore ea laborum voluptate magna eu aliquip mollit
    </span>
    <div
      class="review-author"
      data-testid="author-review"
    >
      Review Card
    </div>
    <div
      class="review-decoration"
    >
      <div
        class="rating -regular -orange"
      >
        <span>
          <i
            class="icon -orange"
          >
            <svg
              height="1em"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M496,203.3H312.36L256,32,199.64,203.3H16L166.21,308.7,107.71,480,256,373.84,404.29,480,345.68,308.7Z"
              />
            </svg>
          </i>
        </span>
        <span>
          <i
            class="icon -orange"
          >
            <svg
              height="1em"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M496,203.3H312.36L256,32,199.64,203.3H16L166.21,308.7,107.71,480,256,373.84,404.29,480,345.68,308.7Z"
              />
            </svg>
          </i>
        </span>
        <span>
          <i
            class="icon -orange"
          >
            <svg
              height="1em"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M496,203.3H312.36L256,32,199.64,203.3H16L166.21,308.7,107.71,480,256,373.84,404.29,480,345.68,308.7Z"
              />
            </svg>
          </i>
        </span>
        <span>
          <i
            class="icon -orange"
          >
            <svg
              height="1em"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M496,203.3H312.36L256,32,199.64,203.3H16L166.21,308.7,107.71,480,256,373.84,404.29,480,345.68,308.7Z"
              />
            </svg>
          </i>
        </span>
        <span>
          <i
            class="icon -orange"
          >
            <svg
              height="1em"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M496,203.3H312.36L256,32,199.64,203.3H16L166.21,308.7,107.71,480,256,373.84,404.29,480,345.68,308.7ZM274.63,347.82,256,334.49V134.39l26,78.91,7.24,22H394.63l-67.32,47.2-19.69,13.81,7.78,22.75,26.26,76.75Z"
              />
            </svg>
          </i>
        </span>
        <span
          class="text -regular -dark"
        >
          4.5
        </span>
      </div>
    </div>
  </div>
</div>
`;
